pkg/controller/servicebinding/reconciler.go

```diff
@@ -165,29 +165,34 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		r.restMapper,
 	)
 	if err != nil {
-		//handle service not found error
-		if k8serrors.IsNotFound(err) {
-			err = updateSBRConditions(r.dynClient, sbr,
-				conditionsv1.Condition{
-					Type:    CollectionReady,
-					Status:  corev1.ConditionFalse,
-					Reason:  ServiceNotFoundReason,
-					Message: err.Error(),
-				},
-				conditionsv1.Condition{
-					Type:   InjectionReady,
-					Status: corev1.ConditionFalse,
-				},
-				conditionsv1.Condition{
-					Type:   BindingReady,
-					Status: corev1.ConditionFalse,
-				},
-			)
-			if err != nil {
-				logger.Error(err, "Failed to update SBR conditions", "sbr", sbr)
+		if sbr.GetDeletionTimestamp() != nil {
+			serviceCtxs = serviceContextList{}
```

Comment:
Wouldn't make sense to execute `buildServiceContexts` only if the `GetDeletionTimestamp() != nil` check fails?